lib/PrincipleStudios.OpenApi.Transformations/SharedOpenApiExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.OpenApi.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PrincipleStudios.OpenApi.Transformations;
+
+public static class SharedOpenApiExtensions
+{
+    public static bool UseOptionalAsNullable(this OpenApiSchema objectSchema, bool useLegacyByDefault = false)
+    {
+        if (!objectSchema.Extensions.TryGetValue("x-ps-optional-as-nullable", out var optionalAsNullable) || optionalAsNullable is not Microsoft.OpenApi.Any.OpenApiBoolean { Value: var result })
```

Comment:
Might be a good idea to put some of these schema options into constants someplace to make them discoverable. Seems like maybe the first of it's kind for now, but something to consider going forward